.github/workflows/documentation.yml

```diff
@@ -59,3 +64,19 @@ jobs:
         if: ${{ github.event_name == 'pull_request' }}
         run: |
             tools/generate_docs.sh build
+  lint:
+    name: Lint documentation
+    needs: documentation_changes
+    if: ${{ needs.documentation_changes.outputs.documentation == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v4
+        with:
+          show-progress: ''
+      - name: markdownlint-cli2-action
+        uses: DavidAnson/markdownlint-cli2-action@v9
+        with:
+          globs: |
+            **/*.md
```

Comment:
Thank you very much for your contribution, we truly appreciate it!

The markdown lint check seems to be setup to check all the markdown files in our repo. By doing this the errors about missing links appear, because the markdown files under `doc/**/*.md` will be copied and pre-processed under the folder `build/doc/**/*.md`. This means we need to find a way to exclude the `doc/**/*.md` from the markdown lint check.

Reply:
How exactly are the link checks excluded?

Reply:
OK, I read the comment above again. 🤣 
We could add the link checking later on, if we see we need it.